### ISSUE TITLE
jQuery-UI-slider still in conflict

### ISSUE THREAD

**Issue by tlartaud** [seiyria/bootstrap-slider#575]
See this comment : https://github.com/seiyria/bootstrap-slider/issues/443#issuecomment-219975383
(wont give any fiddle, sry)

Regards.

**Comment by tlartaud** [comment]
Actually my fix is the following to load this block of just before your script inclusion

```
<script type="text/javascript">
    <!--
    (function ($) {
        "use strict";

        // Fix for seiyria bootstrap slider conflicting with UI-Slider
        $.fn.slider = function () { return true; };

    })(jQuery);
    //-->
</script>
```

For others users, just had this block of code in your page before including bootstrap-slider script and you won't get any problem with ui-slider, and ui-slider will keep working.

Best regards.

**Comment by seiyria** [comment]
Feel free to submit a PR if it's of high importance.

**Comment by tlartaud** [comment]
@seiyria No thanks. Have a lot of work to do, spent a lot of times to debug and report. Feel free to fix your software by yourself. You have the whole solution explained right there.

**Comment by poushy** [comment]
@tlartaud Thank you for this work around.
Same issue happens if bootstrap slider is loaded from the different workflows, if in one of them jquery ui is present, in other it is not.  It is prone to unseen issues, as in one workflow it will initialize as slider and in other as bootstrapSlider.

**Comment by caschbre** [comment]
@tlartaud Thanks for finding this.  With your solution above does it break the other libraries using .slider() ?

**Comment by rovolution** [comment]
@caschbre @tlartaud 

Would it help to have the ability to define your own alternate namespace?

[This is our default behavior](https://github.com/seiyria/bootstrap-slider#what-if-there-is-already-a-slider-plugin-bound-to-the-jquery-namespace), but it seems like the ability to choose the namespace would be valuable? Let me know and I will add it to our "TODO" list.

Thanks!

**Comment by tlartaud** [comment]
@rovolution Hi. Yes that should help... But the really easier solution for everybody should be for you to only bind the namespace : `bootstrapSlider`, because depending on the environment where you deploy (on wordpress frontend vs wordpress backend), you won't get the same librairies and the same inclusions orders. Thats why, sometimes, `slider` will be the one from `jquery-UI`, sometimes not.
I'm really not in favor of such commons names like `slider`.. Because of standards namespacing behaviors, plus, this name is a bit unpersonnal.

I know your default behavior, but its wrong. Its a bad idea to do this.
Just try to create a test page to see the problem in action, with the following scripts included :

**Index.html example 1**
- loading jquery
- loading bootstrapslider.js
- loading jquery-ui-slider.js
- call .slider();

> bootstrapslider is here loaded before jquery-ui-slider, and the function `slider` defined by bootstrapslider will be overrided by jquery-ui-slider. So, in this case, bootstrapslider become completely unavailable, you can't use it.

**Index.html example 2**
- loading jquery
- loading jquery-ui-slider.js
- loading bootstrapslider.js
- call .slider();

> bootstrapslider is here loaded after jquery-ui-slider, so the `slider` function already exists. Bootstrapslider is in this case binding the function `bootstrapSlider` instead of `slider`.

Your idea about giving us the ability to define our alternate namespace is not bad, BUT, it won't fix the first example out of the box. Developpers will still have to understand why, for some reasons, the slider function doesn't return the same thing in some pages, and then, define an alternate namespace. The really best way is to bind a complex namespace by default (bootstrapSlider is okay), and the developper can copy the function to another one with a smaller name if he needs to .

@caschbre I did not test that fix with all librairies, but with jquery-ui-slider, yes, both librairies will keep working.

**Comment by rovolution** [comment]
@tlartaud makes sense. Thanks for reporting the issue.

I am always somewhat hesitant to publish breaking changes, so here is what we might do instead: have the slider always bind to the `bootstrapSlider` namespace and then only bind to the `slider` namespace if none currently exists.

That should address the order of loaded scripts issue described here yet not break the slider for any existing users.

**Comment by tlartaud** [comment]
@rovolution yes, that's the correct thing to do. But you have to keep in mind that it won't fix the issue. Someone who's using your script with the `slider` namespace can still have conflicts, if, in some pages, jQuery-UI-slider is also loaded, because the `slider` namespace will be attributed to jQuery-UI-slider.

To prevent that, you should also recommend in your documentation to use the namespace `bootstrapSlider` instead of `slider`, when loading the script via jQuery, because this namespace is already used by some jQuery libs and could lead into unexpected problems.

Regards.

**Comment by rovolution** [comment]
> Someone who's using your script with the slider namespace can still have conflicts, if, in some pages, jQuery-UI-slider is also loaded, because the slider namespace will be attributed to jQuery-UI-slider (so, for theses pages, he must use bootstrapSlider instead).

I am also going to add a console warning that appears indicating to the user that something is already bound to the slider namespace and that it is recommended that they use the longer namespace.

> To prevent that, you should also recommend in your documentation to use the namespace bootstrapSlider instead of slider, when loading the script via jQuery, because this namespace is already used by some jQuery libs and could lead into unexpected problems.

We will definitely make sure to emphasize that in the docs. thanks!

**Comment by tlartaud** [comment]
The console warning is really a great idea, and can help developpers to not losing time searching everywhere what is going on.

Thanks you to had accepted to discuss about it, and to find the best way to solve it ( I will definitely not say that to your mate @seiyria ... )

Have a nice day,
Cheers.

**Comment by seiyria** [comment]
I don't quite appreciate the snide remarks, as I've mostly held off commenting further in this issue. I still think it's quite simple: don't use two slider libraries. @rovolution is definitely being a great guy here and working towards a compromise, but we differ in opinion here.

**Comment by rovolution** [comment]
Locking this issue as I think its getting a bit off topic :)

**Comment by rovolution** [comment]
published to v9.1.0. see changelog for further details
